compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1363,13 +1339,32 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         let mut collected_lifetimes = FxHashMap::default();
         self.with_hir_id_owner(opaque_ty_node_id, |lctx| {
             let hir_bounds = if origin == hir::OpaqueTyOrigin::TyAlias {
-                lower_bounds(lctx)
+                lctx.lower_param_bounds(bounds, itctx)
             } else {
-                lctx.while_capturing_lifetimes(
-                    opaque_ty_def_id,
-                    &mut collected_lifetimes,
-                    lower_bounds,
-                )
+                debug!(?lctx.captured_lifetimes);
+
+                let lifetime_stash = std::mem::replace(
+                    &mut lctx.captured_lifetimes,
```

Comment:
I don't follow 100% why this field `captured_lifetimes` needs to exist

Reply:
It doesn't this is not 100% finished :). I think it could be merged and this cleaned up in a follow up, but the idea is to remove this. Note the description of the issue.